pytube/contrib/channel.py

```diff
@@ -28,11 +27,15 @@ def __init__(self, url: str, proxies: Optional[Dict[str, str]] = None):
         )
 
         self.videos_url = self.channel_url + '/videos'
+        self.shorts_url = self.channel_url + '/shorts'
```

Comment:
Would you mind adding "Live" as well? `self.live_url = self.channel_url + '/streams'`

```
@property
def live(self) -> Iterable[YouTube]:
    """Yields YouTube objects of live videos in this channel

    :rtype: List[YouTube]
    :returns: List of YouTube
    """
    self._html_page = self.live_url  # Set live tab
    return DeferredGeneratorList(self.videos_generator())

```

Reply:
Supporting multiple tabs is tricky.
I'm using next code as a temporary fix:
```python
try:
    # Possible tabs: Home, Videos, Shorts, Live, Playlists, Community, Channels, About
    active_tab = {}
    for tab in initial_data["contents"]["twoColumnBrowseResultsRenderer"]["tabs"]:
        tab_url = tab["tabRenderer"]["endpoint"]["commandMetadata"]["webCommandMetadata"]["url"]
        if tab_url.rsplit('/', maxsplit=1)[-1] == self.html_url.rsplit('/', maxsplit=1)[-1]:
            active_tab = tab
            break

    # This is the json tree structure for videos
    videos = active_tab["tabRenderer"]["content"]["richGridRenderer"]["contents"]

    # This is the json tree structure of visitor data
    # It is necessary to send the visitorData together with the continuation token
    self._visitor_data = initial_data["responseContext"]["webResponseContextExtensionData"][
        "ytConfigData"]["visitorData"]

except (KeyError, IndexError, TypeError):
```

Reply:
Very interesting your solution, this would prevent `Channel()` from breaking if youtube updated the indexes, and would facilitate the addition of new tabs.

Reply:
Personally I did:
```
tabs = initial_data["contents"]["twoColumnBrowseResultsRenderer"]["tabs"]
videos = next(filter(lambda tab: tab['tabRenderer']['endpoint']['commandMetadata']['webCommandMetadata']['url'].rsplit('/', maxsplit=1)[-1] == 'videos', tabs))
```

Although I think that it could be better to filter based on tab Title instead. Hard to tell which one could be updated by YT more quickly.